src/test/FeeTakingHook.sol

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.20;
+
+import {Hooks} from "../libraries/Hooks.sol";
+import {SafeCast} from "../libraries/SafeCast.sol";
+import {IHooks} from "../interfaces/IHooks.sol";
+import {IPoolManager} from "../interfaces/IPoolManager.sol";
+import {PoolKey} from "../types/PoolKey.sol";
+import {BalanceDelta, toBalanceDelta} from "../types/BalanceDelta.sol";
+import {Currency} from "../types/Currency.sol";
+import {BaseTestHooks} from "./BaseTestHooks.sol";
+
+contract FeeTakingHook is BaseTestHooks {
+    using Hooks for IHooks;
+    using SafeCast for uint256;
+    using SafeCast for int128;
+
+    IPoolManager immutable manager;
+
+    constructor(IPoolManager _manager) {
+        manager = _manager;
+    }
+
+    modifier onlyPoolManager() {
+        require(msg.sender == address(manager));
+        _;
+    }
+
+    uint128 public constant LIQUIDITY_FEE = 543; // 543/10000 = 5.43%
+    uint128 public constant SWAP_FEE_BIPS = 123; // 123/10000 = 1.23%
+    uint128 public constant TOTAL_BIPS = 10000;
+
+    function afterSwap(
+        address, /* sender **/
+        PoolKey calldata key,
+        IPoolManager.SwapParams calldata params,
+        BalanceDelta delta,
+        bytes calldata /* hookData **/
+    ) external override onlyPoolManager returns (bytes4, int128) {
+        // fee will be in the unspecified token of the swap
+        bool specifiedTokenIs0 = (params.amountSpecified < 0 == params.zeroForOne);
```

Comment:
same question as the previous one. Should this be checking the amountSpecified > 0 instead
```suggestion
        bool specifiedTokenIs0 = (params.amountSpecified > 0 == params.zeroForOne);
```

Reply:
Hey! Same answer as @saucepoint above - this PR is built on top of the branch `flip-the-deltas`, where both the currency deltas, and `amountSpecified` have had their sign flipped
Great catch though!

Reply:
I'll definitely update the comment - i missed that!